src/NHibernate/Linq/Visitors/NhPartialEvaluatingExpressionVisitor.cs

```diff
@@ -157,7 +157,9 @@ private Expression EvaluateSubtree(Expression subtree)
 
 		private bool ContainsVariable(Expression expression)
 		{
-			if (!(expression is UnaryExpression unaryExpression))
+			if (!(expression is UnaryExpression unaryExpression) ||
+				// Avoid detecting expression variables as parameters
+				typeof(Expression).IsAssignableFrom(expression.Type.UnwrapIfNullable()))
```

Comment:
`UnwrapIfNullable` looks unnecessary here. Or am I wrong?

Reply:
You're correct. `Expression` is a reference type.

Reply:
Removed.